modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/portlet.js

```diff
@@ -152,6 +177,33 @@
 		},
 	};
 
+	function destroyPortlet(options) {
+
+		// NOTE: Will have to vet all the current listeners to see what
+		// `options` they might care about. A cursory examination suggests that
+		// it is probably only ever (or mostly) `portletId`.
+
+		const {portletId} = options;
+
+		let activeIds = State.readKey(ACTIVE_PORTLET_IDS);
+
+		if (!(portletId in activeIds)) {
+			console.warn(
+				`destroyPortlet(): ${portletId} is not in the set of active portlet IDs`
+			);
+		}
+
+		activeIds = {...activeIds};
```

Comment:
Creating a copy here so that the new ids will be `!==` to the old ids. In general the TS types would complain if we mutated without copying first, although in this case (using `__unsafe__` API), we have to be mindful and do it ourselves (perhaps another reason to pick a target in the "modern" modules and use that as a place to start using `Liferay.State`).

Note that I don't expect that `console.warn()` to ever actually happen, but if it did, making a copy here would mean subscribers still got an update. This maintains parity with the old code, which would fire an event in the same circumstances.